modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectActionVariablesUtil.java

```diff
@@ -58,6 +58,8 @@ public static Map<String, Object> toVariables(
 			).put(
 				"objectDefinitionId",
 				payloadJSONObject.getLong("objectDefinitionId")
+			).put(
+				"classPK", payloadJSONObject.getLong("classPK")
```

Comment:
If the `classPK` is going into the map of variables, it must be defined as a reserved name. Just like that, a user could simply create a field called `classPK` and have their inputs override the actual internal classPK, or vice-versa. Having said that, it's possible that users have already created fields called classPK, so we can't introduce this key anymore.
Now, I'd like you to check two things:
1. Why do the variables inside the map come from the model, when it's a custom object, and from the DTO, when it's a system object? Notice that, if the map of variables were created from the DTO in every case, we would already have the unique "id" key for all cases;
2. If there's actually a need for the above implementation choice, you can just include the key/value "id" (that's already a reserved word) when it's a custom object.

Reply:
Hi @rodrigopaulino 

Yes, i understand the problem with the `classPK`. Follow response about the questions:
1. The _DTO_ referebt the _customObjects_ is received in `payloadJSONObject`, but don't exist _metadata_ for the _customObject_ in `systemObjectDefinitionMetadataTracker`. For this reason, exist the validation for _systemObject_. How implemented now, we can't use the _DTO_ to _system and custom objects_.

2. In this case, do you mean for we use the key `id` in the place the `classPK`?

Reply:
1. The payload comes with both DTO and models of each entry, regardless of their nature, isn't it? If that, why do we populate the map of variables using the model when the object is custom and with the DTO when it's not?
2. Yes. That's the idea. Check if all system objects DTOs come with an "id" and, if they do, only add to the custom objects variables map the key "id" with the value of "classPK". If some system object doesn't have the "id", then make sure to add to all.

Reply:
Hi @rodrigopaulino 

1. So, your proposal is remove the validation if `isSystem` and have a generic method `toVariables` to system and custom object?

2. Ok! So, i do the modification above and map value classPK to key id. Right?

Reply:
@rodrigopaulino 

Looking better, i understand because the map come from the model when it's a custom object and from the DTO when it's a system object.

The data of _systemObjects_ are in DTO. However, the data of _customObjects_ are in `objectEntry` and not in `objectDefinition` (DTO). So, because of this the different treatment. It's make sense?

What i do was refactory the method and map `classPK` to `id`. I'm give push force now.

Reply:
Since we don't know for sure why this choice was made, let's just go with adding the id key/value to all types of objects.